pennylane/ops/qutrit/parametric_ops.py

```diff
@@ -142,27 +144,24 @@ def compute_matrix(
         # The following avoids casting an imaginary quantity to reals when backpropagating
         c = (1 + 0j) * c
         js = -1j * s
-
-        shape = qml.math.shape(theta)
-        is_broadcasted = len(shape) != 0
-        # Construct identity matrices and cast to complex type
-        mat = (
-            qml.math.tensordot([1] * shape[0], qml.math.eye(3), axes=0)
-            if is_broadcasted
-            else qml.math.eye(3)
+        one = qml.math.ones_like(c)
```

Comment:
@Gabriel-Bottrill : explain the reasoning for the implementation; see if other reviewers have suggestions for improvement.